eth_account/_utils/typed_transactions.py

```diff
@@ -345,3 +334,158 @@ def vrs(self) -> Tuple[int, int, int]:
+        cls.assert_valid_fields(dictionary)
+        sanitized_dictionary = pipe(
+            dictionary,
+            dict,
+            partial(merge, cls.transaction_field_defaults),
+            apply_formatters_to_dict(TYPED_TRANSACTION_FORMATTERS),
+        )
+
+        # We have verified the type, we can safely remove it from the dictionary,
+        # given that it is not to be included within the RLP payload.
+        transaction_type = sanitized_dictionary.pop('type')
+        if transaction_type != cls.transaction_type:
+            raise ValueError(
+                "expected transaction type %s, got %s" % (cls.transaction_type, transaction_type),
+            )
+        return cls(
+            dictionary=sanitized_dictionary,
+        )
+
+    @classmethod
+    def from_bytes(cls, encoded_transaction: HexBytes):
+        """Builds a DynamicFeeTransaction from a signed encoded transaction."""
+        if not isinstance(encoded_transaction, HexBytes):
+            raise TypeError("expected Hexbytes, got type: %s" % type(encoded_transaction))
+        if not (len(encoded_transaction) > 0 and encoded_transaction[0] == cls.transaction_type):
+            raise ValueError("unexpected input")
+        # Format is (0x02 || TransactionPayload)
+        # We strip the prefix, and RLP unmarshal the payload into our signed transaction serializer.
+        transaction_payload = encoded_transaction[1:]
+        rlp_serializer = cls._signed_transaction_serializer
+        dictionary = rlp_serializer.from_bytes(transaction_payload).as_dict()  # type: ignore
+        dictionary['type'] = cls.transaction_type
+        return cls.from_dict(dictionary)
+
+    def as_dict(self) -> Dict[str, Any]:
+        """Returns this transaction as a dictionary."""
+        dictionary = self.dictionary.copy()
+        dictionary['type'] = self.__class__.transaction_type
+        return dictionary
+
+    def hash(self) -> bytes:
+        """
+        Hashes this AccessListTransaction to prepare it for signing.
+        As per the EIP-2930 specifications, the signature is a secp256k1 signature over
+        keccak256(0x01 || rlp([chainId, nonce, gasPrice, gasLimit, to, value, data, accessList])).
+        Here, we compute the keccak256(...) hash.
+        """
+        # Remove signature fields.
+        transaction_without_signature_fields = dissoc(self.dictionary, 'v', 'r', 's')
+        rlp_serializer = self.__class__._unsigned_transaction_serializer
+        hash = pipe(
+            rlp_serializer.from_dict(transaction_without_signature_fields),  # type: ignore
+            lambda val: rlp.encode(val),  # rlp([...])
+            lambda val: bytes([self.__class__.transaction_type]) + val,  # (0x02 || rlp([...]))
+            keccak,  # keccak256(0x02 || rlp([...]))
+        )
+        return cast(bytes, hash)
+
+    def payload(self) -> bytes:
+        """
+        Returns this transaction's payload as bytes. Here, the TransactionPayload = rlp([chainId,
```

Comment:
Same comment as above on the `rlp()` part here. We can do without `accessList` and `gasPrice` in the example and refactor to reflect EIP-1559 transactions.